### PR TITLE
Blobs use canpass to check if they can expand instead of a density check

### LINKED ISSUE #14991: Blobs cannot expand onto a tile using the verb if there's a blob spore there
It hits the blob spore and doesn't expand. Very bad.

### PR DESCRIPTION
**Pull request by ChangelingRain**
Functionality is the same(I tested it, even!)
Fixes #14991

### CODE REVIEW

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 1]
```diff
@@ -21,6 +21,11 @@
 /mob/living/simple_animal/hostile/blob/blob_act()
 	return
 
+/mob/living/simple_animal/hostile/blob/CanPass(atom/movable/mover, turf/target, height=0)
+	if(istype(mover, /obj/effect/blob))
```
Comment (neersighted) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: Don't we have isblob or something?

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 2]
```diff
@@ -21,6 +21,11 @@
 /mob/living/simple_animal/hostile/blob/blob_act()
 	return
 
+/mob/living/simple_animal/hostile/blob/CanPass(atom/movable/mover, turf/target, height=0)
```
Comment (neersighted) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: Spaces in default parameters. However, if you don't use them, not need to specify them (`..()` will call correctly without them).

File: code/game/gamemodes/blob/theblob.dm [hunk 3]
```diff
@@ -132,16 +132,19 @@
 	if(!T)
 		return 0
 	var/make_blob = 1 //can we make a blob?
+
 	if(istype(T, /turf/space) && prob(65))
 		make_blob = 0
```
Comment (neersighted) [inline code/game/gamemodes/blob/theblob.dm]: Use TRUE/FALSE defines?

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: nope

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: this is not my fault and I don't care enough about it to fix it

File: code/game/gamemodes/blob/theblob.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/theblob.dm]: could do that, but it'd be pretty minor and mostly just a style change

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: more accurately, I care, but my care is all hate and this isn't the PR to remove the height arg from CanPass entirely

Comment (optimumtact) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: fix it or your PR isn't getting merged

File: code/game/gamemodes/blob/theblob.dm (hunk 3 above)
Comment (optimumtact) [inline code/game/gamemodes/blob/theblob.dm]: Fix it

**Pull request merged**